Source/Scene/PointPrimitive.js

```diff
@@ -0,0 +1,488 @@
+        }
+
+        Matrix4.multiplyByPoint(modelMatrix, position, tempCartesian3);
+        return SceneTransforms.computeActualWgs84Position(frameState, tempCartesian3);
+    };
+
+    var scratchMatrix4 = new Matrix4();
+    var scratchCartesian4 = new Cartesian4();
+
+    PointPrimitive._computeScreenSpacePosition = function(modelMatrix, position, scene) {
+        // This function is basically a stripped-down JavaScript version of PointPrimitiveCollectionVS.glsl
+        var camera = scene.camera;
+        var view = camera.viewMatrix;
+        var projection = camera.frustum.projectionMatrix;
+
+        // Model to eye coordinates
+        var mv = Matrix4.multiplyTransformation(view, modelMatrix, scratchMatrix4);
+        var positionEC = Matrix4.multiplyByVector(mv, Cartesian4.fromElements(position.x, position.y, position.z, 1, scratchCartesian4), scratchCartesian4);
+        var positionCC = Matrix4.multiplyByVector(projection, positionEC, scratchCartesian4); // clip coordinates
+        var positionWC = SceneTransforms.clipToGLWindowCoordinates(scene, positionCC, new Cartesian2());
+
+        return positionWC;
+    };
+
+    /**
+     * Computes the screen-space position of the pointPrimitive's origin.
+     * The screen space origin is the top, left corner of the canvas; <code>x</code> increases from
+     * left to right, and <code>y</code> increases from top to bottom.
+     *
+     * @param {Scene} scene The scene.
+     * @returns {Cartesian2} The screen-space position of the pointPrimitive.
+     *
+     * @exception {DeveloperError} PointPrimitive must be in a collection.
+     *
+     * @example
+     * console.log(p.computeScreenSpacePosition(scene).toString());
+     */
+    PointPrimitive.prototype.computeScreenSpacePosition = function(scene) {
+        var pointPrimitiveCollection = this._pointPrimitiveCollection;
+
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(pointPrimitiveCollection)) {
+            throw new DeveloperError('PointPrimitive must be in a collection.');
+        }
+        if (!defined(scene)) {
+            throw new DeveloperError('scene is required.');
+        }
+        //>>includeEnd('debug');
+
+        var modelMatrix = pointPrimitiveCollection.modelMatrix;
+        var windowCoordinates = PointPrimitive._computeScreenSpacePosition(modelMatrix, this._actualPosition, scene);
+        windowCoordinates.y = scene.canvas.clientHeight - windowCoordinates.y;
+        return windowCoordinates;
+    };
+
+    /**
+     * Determines if this pointPrimitive equals another pointPrimitive.  PointPrimitives are equal if all their properties
+     * are equal.  PointPrimitives in different collections can be equal.
+     *
+     * @param {PointPrimitive} other The pointPrimitive to compare for equality.
+     * @returns {Boolean} <code>true</code> if the pointPrimitives are equal; otherwise, <code>false</code>.
+     */
+    PointPrimitive.prototype.equals = function(other) {
```

Comment:
Is this a common operation?

I suspect it (and `Billboard`, `Label`, etc) would be a faster in the reject case if we move this to above testing show:
```javascript
Cartesian3.equals(this._position, other._position) &&
```